packages/redux/src/themes/__tests__/reducer.test.ts

```diff
@@ -0,0 +1,156 @@
+import * as actionTypes from '../actionTypes.js';
+import { mockTheme, mockThemeCode } from 'tests/__fixtures__/themes/index.mjs';
+import { toBlackoutError } from '@farfetch/blackout-client';
+import reducer, { INITIAL_STATE } from '../reducer.js';
+import type { ThemeState } from '../types/index.js';
+
+const mockAction = { type: 'foo' };
+const meta = { code: mockThemeCode };
+let initialState: ThemeState;
+
+describe('themes reducer', () => {
+  beforeEach(() => {
+    initialState = reducer(INITIAL_STATE, mockAction);
+  });
+
+  describe('error() reducer', () => {
+    const error = 'An error occurred';
+    const expectedError = {
+      [mockThemeCode]: error,
+    };
+
+    it('should return the initial state', () => {
+      const state = reducer(INITIAL_STATE, mockAction).error;
+
+      expect(state).toEqual(initialState.error);
+    });
+
+    it(`should handle ${actionTypes.FETCH_THEME_REQUEST} action type`, () => {
+      const expectedError = {
+        [mockThemeCode]: undefined,
+      };
+      const state = reducer(undefined, {
```

Comment:
Please, use a different initial state than `undefined` to make sure the test is testing the correct code. Also the test is not validating if `isLoading` is true.

Reply:
(Disregard the comment regarding `isLoading`, this will only test the `error` sub reducer.)